### PR TITLE
Vendor k8s 1.6 updates

### PR DESCRIPTION
**Pull request by chrislovecnm**
This PR updates vendoring needed to build kops against kubernetes 1.6 code base.

This is still a work in progress:

- [x] protokube has issues with the code marshaling the Pod type
- [ ] more testing needed
- [ ] test dnscontroller
- [x] test protokube
- [ ] api machinery make needs to be fixed.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2068)

### CODE REVIEW

File: protokube/pkg/protokube/codecs.go [hunk 1]
```diff
@@ -20,14 +20,15 @@ import (
...
 	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/api/v1"
-	"k8s.io/kubernetes/pkg/runtime"
 
 	_ "k8s.io/kubernetes/pkg/api/install"
 )
 
 func encoder() runtime.Encoder {
+	// This encoder is kubernetes core, not kops encoder
 	yaml, ok := runtime.SerializerInfoForMediaType(api.Codecs.SupportedMediaTypes(), "application/yaml")
```
Comment (chrislovecnm) [inline protokube/pkg/protokube/codecs.go]: @lavalamp this previously worked, and I am thinking that we may need to initialize both client-go and kops apis.

```
--- FAIL: TestBuildEtcdManifest (0.00s)
	integration_test.go:72: error marshalling to yaml: error encoding *v1.Pod: no kind is registered for the type v1.Pod
```

This is the error we are getting, and I am not certain if we need the kops api here as well.  @justinsb does protokube utilize types from the kops api?

@lavalamp if this code requires both the Pod type and kops api types - ideas on how to pull that on off?

File: cmd/kops/create.go [hunk 2]
```diff
@@ -80,7 +79,7 @@ func RunCreate(f *util.Factory, out io.Writer, c *CreateOptions) error {
 	}
 
 	// Codecs provides access to encoding and decoding for the scheme
-	codecs := k8sapi.Codecs //serializer.NewCodecFactory(scheme)
+	codecs := kopsapi.Codecs //serializer.NewCodecFactory(scheme)
```
Comment (justinsb) [inline cmd/kops/create.go]: Switching from k8sapi -> kopsapi is suspicious.  But I think kopsapi is right, so ...

File: pkg/client/simple/vfsclientset/import_known_versions.go [hunk 3]
```diff
@@ -20,13 +20,24 @@ package vfsclientset
...
+	_ "k8s.io/client-go/pkg/apis/authentication/install"
+	_ "k8s.io/client-go/pkg/apis/authorization/install"
+	_ "k8s.io/client-go/pkg/apis/autoscaling/install"
+	_ "k8s.io/client-go/pkg/apis/batch/install"
+	_ "k8s.io/client-go/pkg/apis/certificates/install"
+	_ "k8s.io/client-go/pkg/apis/extensions/install"
+	_ "k8s.io/client-go/pkg/apis/policy/install"
+	_ "k8s.io/client-go/pkg/apis/rbac/install"
+	_ "k8s.io/client-go/pkg/apis/settings/install"
+	_ "k8s.io/client-go/pkg/apis/storage/install"
```
Comment (justinsb) [inline pkg/client/simple/vfsclientset/import_known_versions.go]: I guess they split this up so we don't need to import everything.

But ... also the kops import is missing now?

File: cmd/kops/create.go (hunk 2 above)
Comment (chrislovecnm) [inline cmd/kops/create.go]: that is part of the new changes to apimachinery ... and figuring it out was not fun.

File: pkg/client/simple/vfsclientset/import_known_versions.go (hunk 3 above)
Comment (chrislovecnm) [inline pkg/client/simple/vfsclientset/import_known_versions.go]: Added kops back in, but I think it is being initialized elsewhere.

File: protokube/pkg/protokube/codecs.go (hunk 1 above)
Comment (chrislovecnm) [inline protokube/pkg/protokube/codecs.go]: I think I fixed it ... but @justinsb do we need the codec to parse kops api types as well?

**Comment by chrislovecnm** [comment]
@k8s-bot aws e2e test this

**Comment by chrislovecnm** [comment]
@k8s-bot aws e2e test this

**Comment by chrislovecnm** [comment]
@zmerlynn I may need some help debugging this ;(  I am able to do a full test run on my end with my own nodeup, protokube, etc.

**Comment by chrislovecnm** [comment]
@justinsb this is done and tested.   Just rebased.  I am not sure why e2e is failing.  Ideas?

**Comment by justinsb** [comment]
That's why I did the similar branch, as it seems the only way to actually review this PR.

**Comment by justinsb** [comment]
OK, #2128 is passing e2e now.  The diffs are:

* The git submodules don't match, I think this is mostly benign but you don't need all the new volume drivers if you go with a later version (see #2127 for a helper utility)
* The KubernetesClient initialization has changed.  I'm surprised you were able to bring up a cluster, actually.  The new initialization looks like this: https://github.com/kubernetes/client-go/issues/149  This block occurs in channels/cmd/channels/root.go, cmd/kops/rollingupdatecluster.go, cmd/kops/validate_cluster.go, protokube/pkg/protokube/kube_context.go
* You need to re-enable apimachinery and re-run the conversion

Those are the primary differences AFAICT.  I think then e2e will pass.  You can pull both branches and do a diff though.

I'd prefer to merge this PR if you can get these things fixed up - it seems fair.

**Pull request closed without merging**